src/streams/getIsNotificationEnabled.js

```diff
@@ -0,0 +1,7 @@
+/* @flow strict-local */
+import type { Subscription } from '../types';
+
+export default (
+  subscription: Subscription | void,
+  userSettingStreamNotification: boolean,
+): boolean => subscription?.push_notifications ?? userSettingStreamNotification;
```

Comment:
Cool, this solves the problem.

Some nits I would do differently:
* This is really about a subscription (the user's relationship to a stream, represented by a `Subscription` value) rather than a stream itself (represented by `Stream`), so I'd put it in `src/subscriptions/` or perhaps `src/settings/`. (This is a distinction our existing code isn't super consistent about.)
* I've found it helpful to shift toward fewer, medium-size files rather than lots of tiny files. So instead of giving this its own file, I would make it the first thing added to a file named `src/subscriptions/subscriptionsModel.js` ~~or `src/settings/settingsModel.js`~~ (definitely "subscriptions"; see below). See b21d7c58de80f40b15ca7de68c2575c5824fd9de (`muteModel`) for a recent example that's equally tiny for now. See `src/unread/unreadModel.js` for a larger example, with the reducer in the same file; this function would be a "getter", a lot like the function `getUnreadCountForTopic` (added in 0b4c2a446a62d424456cf61005cb23db8876f184) there.
* I think my ideal version of this function would take the whole `PerAccountSettingsState` object, i.e. the return value of `getSettings`, rather than this one flag from it. That way the details of which piece of the user settings is relevant are internal to this function. That also means that where we have code which is ultimately going to call both this function and a few more like it, that code only needs to pass around a single settings object, and doesn't have to keep adding more bits of settings to pass down through the chain each time it has to call another function like this one.
  * That is *very* recent, though -- I might not have felt the same way before splitting up `PerAccountSettingsState` vs `GlobalSettingsState`, which just happened in 3d98cf9c405da325ae740ea58b79c9fb749fbcc3 / #5017 a couple of weeks ago (and after you wrote this revision.) In any case I definitely haven't discussed or demonstrated this idea much in the past.
  * Also this may involve a bit of refactoring in the layers above where this gets called. Like I said, I think this will lead to less churn there as we add more functions like this, but it means a bit more right now.
* I'd like the name to somehow indicate that it's per-stream, as contrasted with something like `state.settings.offlineNotification` which feels like this same name `getIsNotificationEnabled` could describe it too. The types already mean there isn't a great danger of outright mixing this up with something that isn't per-stream, but still it'd be good to make it more immediately obvious to the reader.
  * Something I quite like about this name is that it makes clear it's getting an on/off setting! The first few names I think of that do indicate it's per-stream, plus align with the property name `push_notifications`, are like `getStreamPushNotifications` and they sound like they're getting some actual notifications, whatever that would mean. A name that has both these good properties will probably be somewhat wordy, but I think that's OK.

All of those are things I'm happy to take care of after we merge this PR, though -- no need to revise them here. This already solves the really key problem, which is that I want this logic in one place and with its own name.

---

Oh, but one pair of things I think I would like to have before merge:
* A bit of jsdoc saying what this is intended to mean.
* A jsdoc comment on the `push_notifications` property where `Subscription` is defined, saying to use this function to interpret it.

Those are key because the problem this is meant to solve is, when someone goes to use that property (just as you did in the original revision of this PR), we want the easy natural thing to be that they spot this function and use it so that they get the right behavior. So the comment from the property to help you find this function is a key part of that.

Hmm, and I guess thinking about that answers the question I was waffling on above about whether this goes in a "subscription" model or "settings" model: it should go in a `subscriptionsModel.js`, because this function is something you need in order to correctly interpret the subscriptions data. Or from another angle, the question this function is an answer to is a question that comes up when you're thinking about a subscription, not when you're thinking about settings.